routing/cross_routing_context.hpp

```diff
@@ -70,6 +70,8 @@ struct OutgoingCrossNode
   void Save(Writer & w) const;
 
   size_t Load(Reader const & r, size_t pos, size_t adjacencyIndex);
+
+  m2::RectD const GetLimitRect() const { return m2::RectD(m_point.lat, m_point.lon, m_point.lat, m_point.lon); }
```

Comment:
Yes. Now it's used this way in IngoingCrossNode class. So if to refactor it should be refactored in two cases anyway.

m_ingoingIndex and m_outgoingIndex declared in CrossRoutingContextReader.
```
  m4::Tree<IngoingCrossNode> m_ingoingIndex;
  m4::Tree<OutgoingCrossNode> m_outgoingIndex;
```
As far as I'm aware to work correctly IngoingCrossNode and OutgoingCrossNode have to have method ```m2::RectD const GetLimitRect() const;```.  It's possible to convert to mercator in GetLimitRect() before returning. There's another possibility. To keep m_point in IngoingCrossNode and OutgoingCrossNode in mercator. But in IngoingCrossNode and OutgoingCrossNode PointD and RectD are not in mercator in several places.  For example,
```
void OutgoingCrossNode::Save(Writer & w) const
{
  uint64_t point = PointToInt64(m2::PointD(m_point.lon, m_point.lat), kCoordBits);
...
```

On the other hand this class is about to be removed. I'm not sure that it's worth refactoring them now.